### PR TITLE
Longest Substring Without Repeating Characters

### PR DESCRIPTION
**Pull request by piyushgoell**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

C++ Implementation of Longest Substring Without Repeating Characters

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Longest Substring Without Repeating Characters #563

### CODE REVIEW

**Comment by piyushgoell** [comment]
Please review this PR @SKAUL05

**Comment by piyushgoell** [comment]
@SKAUL05

**Comment by piyushgoell** [comment]
@SKAUL05

File: C-Plus-Plus/string/LongestSubstringWithoutRepeatingCharacters.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  int i = 0, res = 0;
+  for (int j = 0; j < s.length(); j++)
+  {
+    i = max(i, prev[s[j]] + 1);
+    int maxEnding = j - i + 1;
+    res = max(res, maxEnding);
+    prev[s[j]] = j;
+  }
+  return res;
+}
```
Comment (HarshCasper) [inline C-Plus-Plus/string/LongestSubstringWithoutRepeatingCharacters.cpp:21]: Add sample Input/Output

File: C-Plus-Plus/string/LongestSubstringWithoutRepeatingCharacters.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#include <algorithm>
```
Comment (HarshCasper) [inline C-Plus-Plus/string/LongestSubstringWithoutRepeatingCharacters.cpp:1]: Describe the Problem

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Pull request closed without merging**